pandas/core/arrays/base.py

```diff
@@ -406,8 +410,23 @@ def argsort(self, ascending=True, kind='quicksort', *args, **kwargs):
         # 1. _values_for_argsort : construct the values passed to np.argsort
         # 2. argsort : total control over sorting.
         ascending = nv.validate_argsort_with_ascending(ascending, args, kwargs)
-        values = self._values_for_argsort()
-        result = np.argsort(values, kind=kind, **kwargs)
+        values, mask = self._values_for_argsort()
+
+        if mask.any():
+            notmask = ~mask
+            notnull = np.argsort(values[notmask], kind=kind, **kwargs)
+
+            # permu maps the indices of the subarray
+            # without nan to the indices of the original array.
+            permu = np.arange(len(mask))
```

Comment:
where is the name "permu" coming from? 
Maybe have a look in pd.core.sorting.nargsort to use more or less the same names in the code.